src/utils/helpers.py

```diff
@@ -1,14 +1,35 @@
 import json
 import logging
+import shutil
 from pathlib import Path
 
 import yaml
 
 PATH_SPARK_LOG = "/tmp/logs/spark.log"
 
 
+def clear_directory(dir: Path | str) -> None:
+    """
+    Clears all files and subdirectories in the specified directory, except
+    for .gitkeep files.
+
+    Args:
+        dir: A string or Path object to the target directory, relative to
+            the Docker container.
+    """
+    dir_path = Path(dir)
+    if not dir_path.is_dir():
+        raise ValueError(f"The path {dir} is not a valid directory.")
+
+    for item in dir_path.iterdir():
+        if item.is_dir():
+            shutil.rmtree(item.as_posix())
+        elif item.name != ".gitkeep":
+            item.unlink()
+
+
 def create_python_logger(
-    name: str, log_file_path: str = PATH_SPARK_LOG
+    name: str, mode: str = "a", log_file_path: str = PATH_SPARK_LOG
```

Comment:
**[Thought, non-blocking]** I wonder if there's a way we can accomplish the same goal without having to rely on the assumption that certain parts of the code will run before others? Not a huge deal, if it works we can always iterate on it later.

Reply:
In this case, it's not actually too bad if the code happened to run out-of-order, as we're basically only using the write mode to clear the previous file. If stuff ran out-of-order somehow, we'd just end up appending to the file instead of truncating it first.

I tried deleting it manually before writing, but Python gets mad when the file it was writing to ceases to exist _after_ the logger has started.